export_dagmc_cmd/mcnp2cad.cpp

```diff
@@ -0,0 +1,1325 @@
+}
+
+std::vector<std::string> MCNP2CAD::get_syntax_help()
+{
+  std::vector<std::string> help;
+  return help;
+}
+
+std::vector<std::string> MCNP2CAD::get_help()
+{
+  std::vector<std::string> help;
+  return help;
+}
+
+void debugSurfaceDistances( InputDeck& deck, std::ostream& out = record ){
+
+  InputDeck::surface_card_list& surfaces = deck.getSurfaces();
+  int k = 0;
+  int facet, ident;
+  for( InputDeck::surface_card_list::iterator i = surfaces.begin(); i!=surfaces.end(); ++i){
+    if( surfaces.at(k)->getIdent() < 0 ){
+      ident = surfaces.at(k)->getIdent()/10;
+      facet = -( surfaces.at(k)->getIdent() - 10*ident );
+    }
+    else{
+      facet = 0;
+    }
+    k++;
+    try{
+
+      //Facets are numbered as -10*(identification).
+      const SurfaceVolume& s = makeSurface(*i, NULL, facet );
+      out << "S" << (*i)->getIdent() << " distance from origin: " << s.getFarthestExtentFromOrigin() << std::endl;
+    }
+    catch(std::runtime_error& e){
+      record << "Error debugging surface distances: ";
+      std::cerr << "Error debugging surface distances: ";
+      throw;
+    }
+  }
+  
+}
+
+
+bool MCNP2CAD::execute(CubitCommandData &data)
+{
+
+  record.open("mcnp_import.log");
+
+  std::string mcnp2cad_version(bool full = true);
+
+
+  std::string filename;
+  data.get_string("filename",filename);
+
+  bool rval;
+
+  rval = parse_options(data);
+
+//  TODO: find a way to check whether the parsing worked as expected
+//  CHK_MB_ERR_RET("Error parsing options: ",rval);
+
+
+  try{
+    std::ifstream input( filename.c_str(), std::ios::in );
+    InputDeck& deck = InputDeck::build(input);
+    record << "filename is " << filename << std::endl;
+  
+    // turn off debug if it was set by debug_input only
+    if( din ){ Gopt.debug = false; }
+    
+    if( dout && !OPT_DEBUG ){ Gopt.debug = true; }
+  
+    if( OPT_DEBUG ){ 
+      debugSurfaceDistances( deck );
+    }
+  
+    iGeom_Instance igm;
+    int igm_result;
+  
+    iGeom_newGeom( Gopt.igeom_init_options.c_str(), &igm, &igm_result, Gopt.igeom_init_options.length() );
+    CHECK_IGEOM( igm_result, "Initializing iGeom" );
+  
+    GeometryContext context( igm, deck );
+    context.createGeometry();
+  }
+  catch(std::runtime_error& e){
+    record << e.what() << std::endl;
+    PRINT_INFO( e.what() );
+    PRINT_INFO( "\n" );
+    std::cerr << e.what() << std::endl;
+    record.close();
+    return false;
+  }
+  
+  record.close();
+  return true;
+}
+
+bool MCNP2CAD::parse_options(CubitCommandData &data)
+{
+
+  // read parsed command for tolerance
+  data.get_value("specific_tolerance",Gopt.specific_tolerance);
+  record << "Setting specific tolerance to " << Gopt.specific_tolerance << std::endl;
+  //If tolerance was specified
+  if( data.find_keyword("tol") ){
+    Gopt.override_tolerance = true;
+    if( Gopt.specific_tolerance <= 0.0 || Gopt.specific_tolerance > .1 ){
+      std::cerr << "Warning: you seem to have specified an unusual tolerance (" 
```

Comment:
My understanding is that the std::cerr is going to print to the terminal rather than the Trelis console. If we want users to see this message, we should probably provide it using the PRINT_INFO macro provided by the Trelis SDK.